File: codes/java/chap06/answer_A41.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		// 입력
		int N = sc.nextInt();
		String S = sc.next();

		// 답을 구한다
		boolean Answer = false;
		for (int i = 0; i <= N - 3; i++) {
			if (S.charAt(i)=='R' && S.charAt(i+1)=='R' && S.charAt(i+2)=='R') Answer = true;
			if (S.charAt(i)=='B' && S.charAt(i+1)=='B' && S.charAt(i+2)=='B') Answer = true;
		}

		// 출력
		if (Answer == true) System.out.println("Yes");
		else System.out.println("No");
	}
};

File: codes/cpp/chap06/answer_A41.cpp
#include <iostream>
using namespace std;

int main() {
	// 입력
	int N; char S[200009];
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> S[i];

	// 답을 구한다
	bool Answer = false;
	for (int i = 1; i <= N - 2; i++) {
		if (S[i] == 'R' && S[i + 1] == 'R' && S[i + 2] == 'R') Answer = true;
		if (S[i] == 'B' && S[i + 1] == 'B' && S[i + 2] == 'B') Answer = true;
	}

	// 출력
	if (Answer == true) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}
